docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,272 @@
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+</project>
+-------------
+
+The key points are:
+
+* By conventon, the deployment module has the `-deployment` suffix (`quarkus-smallrye-health-deployment`).
+* The deployment module depends on the `quarkus-core-deployment`. We will see later which dependencies are convenient to add.
+* We add  the `quarkus-extension-processor` to the compiler annotation processors.
+
+=== The Runtime module
+[source, shell]
+---------------
+$ mkdir -p ./runtime/src/main/java/
+---------------
+Let's create the runtime's `pom.xml`.
+[source, xml]
+-------------
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-smallrye-health-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-smallrye-health</artifactId>
+    <name>Quarkus - SmallRye Health - Runtime</name>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-core</artifactId>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-bootstrap-maven-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.quarkus</groupId>
+                            <artifactId>quarkus-extension-processor</artifactId>
+                            <version>${project.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
+-------------
+
+The key points are:
+
+* By convention, the runtime module has no suffix (`quarkus-smallrye-health`).
+* The deployment module depends on the `quarkus-core`. We will see later which dependencies are convenient to add.
+* We add  the `quarkus-bootstrap-maven-plugin` to generate the Quarkus extension descriptor included into the runtime artifact.
+* We add  the `quarkus-extension-processor` to the compiler annotation processors.
+
+=== Registering your extension.
+At this point, your extension is almost initialized. You need to declare your sub-modules to be used by other extensions. You should add to the `quarkus/build-parent/pom.xml`.
```

Comment:
Ah, you mean that the extension is intended to be a part of the Quarkus repo here? I think that complicate the setup. It could be easier to start developing an extension as a separate project, imo.

Reply:
I think both have values.

It should probably be 2 different sections. The thing with an external extension is that we do not have a proper "deployment" bom right now. I know Daniel has plans for creating a Maven mojo to create an extension so it should simplify the setup a lot.